fix(tile): grow tiles along cross-axis

bennypowers opened the pull request:
Fixes #1359 

## What I did

1. make tile internal containers grow in the cross-axis, so that the host can flex as well. Make the host a flex container


## Testing Instructions

1. compare *all* the DP demos to production

brianferry commented:
I see this also looks like it fixes the problem on the checkable demo where the tile takes up the full width instead of the 320px width it's set to on the page.  

Color context demo I am seeing the arrow now on the edge of the tile anymore

![image](https://github.com/RedHat-UX/red-hat-design-system/assets/9708152/991e5f34-f77d-4ae5-81d3-ca0b23f882a8)

Seeing the same on compact demo.

brianferry commented:
Outside of the arrow issue, everything looks good.  It looks like setting the rh-tile width to be calc(10 * --rh-space-2xl) also fixes the bugs demo but it is not included there originally.  Would that be enough to fix the issue in general or do we need this PR for that fix too?

marionnegp reviewed (changes requested):
Under the Tile and Checkable demos, I'm able to select the disabled checkbox tile, which I don't think should be possible.

I'm seeing the same issue with the compact tile demos as @brianferry. 

Under the color context demo, the footer doesn't stick to the bottom in the basic card without an image.
![Screenshot 2023-11-28 at 10 21 08 AM](https://github.com/RedHat-UX/red-hat-design-system/assets/95588923/4c8e1ec6-d439-4e24-bb2b-e9e6ba8411f7)

Under the Full Width Images demo, the headline slot should be using `rh-color-interactive-blue-darker` and shouldn't be underlined in its default state.
![Screenshot 2023-11-28 at 10 23 17 AM](https://github.com/RedHat-UX/red-hat-design-system/assets/95588923/30a23aa2-6e06-44f8-b3a3-975198e81db8)

Under the Tile Group Disabled and Tile Group Radio Disabled demos, should disabled tiles be focusable? I'm able to focus on the first tile only.

bennypowers commented:
@marionnegp see #1302 for those issues

brianferry commented on elements/rh-tile/demo/tile-group-flex.html:
```diff
@@ -0,0 +1,36 @@
+    <h2 slot="headline">
+      <a href="#top">Link</a>
+    </h2>
+    Lorem ipsum dolor sit amet, consectetur adipiscing elit.
```
Resolve to merge

brianferry reviewed (approved):
Code looks good, fixed the arrow issue I was seeing earlier.  👍

bennypowers commented:
@marionnegp I addressed the layout issue with footer (full-height content)
and the issue of the headline link (that demo did not add the lightdom css, which I added back in just now)

The other issues should be fixed by #1302

marionnegp reviewed (commented):
Thanks, @bennypowers! Lauded Gold Titleholder Mopes
